pkg/driver/oxide/oxide.go

```diff
@@ -0,0 +1,423 @@
+		mcnflag.StringFlag{
+			Name:   flagHost,
+			Usage:  "Oxide host",
+			EnvVar: "OXIDE_HOST",
+		},
+		mcnflag.StringFlag{
+			Name:   flagToken,
+			Usage:  "Oxide token",
+			EnvVar: "OXIDE_TOKEN",
+		},
+
+		mcnflag.StringFlag{
+			Name:   flagProject,
+			Usage:  "Oxide project",
+			EnvVar: "OXIDE_PROJECT",
+		},
+		mcnflag.StringFlag{
+			Name:   flagUserData,
+			Usage:  "Oxide user data (cloud init)",
+			EnvVar: "OXIDE_USER_DATA",
+		},
+		mcnflag.StringFlag{
+			Name:   flagExtraUserData,
+			Usage:  "Oxide extra user data (cloud init)",
+			EnvVar: "OXIDE_EXTRA_USER_DATA",
+		},
+
+		mcnflag.IntFlag{
+			Name:   flagDiskSize,
+			Usage:  "Oxide disk size",
+			EnvVar: "OXIDE_DISK_SIZE",
+			Value:  34359738368,
+		},
+		mcnflag.StringFlag{
+			Name:   flagDiskImageId,
+			Usage:  "Oxide disk image id",
+			EnvVar: "OXIDE_DISK_IMAGE_ID",
+			Value:  "c1bc5374-614b-479e-8352-ac4526c29f97",
+		},
+		mcnflag.IntFlag{
+			Name:   flagMemory,
+			Usage:  "Oxide memory",
+			EnvVar: "OXIDE_MEMORY",
+			Value:  4294967296,
+		},
+		mcnflag.IntFlag{
+			Name:   flagNCpus,
+			Usage:  "Oxide Number of CPUs",
+			EnvVar: "OXIDE_NUMBER_OF_CPUS",
+			Value:  2,
+		},
+
+		mcnflag.StringFlag{
+			Name:   flagSSHUser,
+			Usage:  "Oxide user",
+			EnvVar: "OXIDE_SSH_USER",
+			Value:  defaultSSHUser,
+		},
+		mcnflag.StringSliceFlag{
+			Name:   flagSSHPubKeys,
+			Usage:  "Oxide ssh public key IDs",
+			EnvVar: "OXIDE_SSH_PUB_KEYS",
+			Value:  []string{"34f34227-cf76-4b1e-85ae-17ecfc744f00"},
+		},
+
+		mcnflag.StringFlag{
+			Name:   flagVpc,
+			Usage:  "Oxide vpc",
+			EnvVar: "OXIDE_VPC",
+			Value:  "default",
+		},
+		mcnflag.StringFlag{
+			Name:   flagSubnet,
+			Usage:  "Oxide subnet",
+			EnvVar: "OXIDE_SUBNET",
+			Value:  "default",
+		},
+	}
+}
+
+func (d *Driver) SetConfigFromFlags(opts drivers.DriverOptions) (err error) {
+	flags := []struct {
+		target *string
+		flag   string
+	}{
+		{&d.Host, flagHost},
+		{&d.Token, flagToken},
+		{&d.Project, flagProject},
+	}
+
+	for _, f := range flags {
+		*f.target = opts.String(f.flag)
+		if *f.target == "" {
+			return requiredOptionError(f.flag)
+		}
+	}
+
+	d.Host = opts.String(flagHost)
+	d.Token = opts.String(flagToken)
+	d.Project = opts.String(flagProject)
+	d.UserData = opts.String(flagUserData)
+	d.ExtraUserData = opts.String(flagExtraUserData)
+
+	d.DiskSize = opts.Int(flagDiskSize)
+	d.DiskImageId = opts.String(flagDiskImageId)
+	d.Memory = opts.Int(flagMemory)
+	d.Ncpus = opts.Int(flagNCpus)
+
+	d.VpcName = opts.String(flagVpc)
+	d.SubnetName = opts.String(flagSubnet)
+
+	d.BaseDriver.SSHUser = opts.String(flagSSHUser)
+	d.SSHKeys = opts.StringSlice(flagSSHPubKeys)
+
+	// base driver flags
+	d.BaseDriver.SSHPort = 22
+	d.SetSwarmConfigFromFlags(opts)
+
+	return
+}
+
+// PreCreateCheck is called to enforce pre-creation steps
+func (d *Driver) PreCreateCheck() error {
+	if len(d.userDataBase64())*3/4 > 32767 {
+		return fmt.Errorf("user data is too large")
```

Comment:
Would be really helpful for the user to know what the limit is as part of the error message

Reply:
added to return value